src/GafferSceneUI/SetMembershipInspector.cpp

```diff
@@ -0,0 +1,222 @@
+		);
+	},
+	// Max cost
+	1000,
+	// Removal callback
+	[] ( const HistoryCacheKey &key, const SceneAlgo::History::ConstPtr &history ) {
+		// Histories contain PlugPtrs, which could potentially be the sole
+		// owners. Destroying plugs can trigger dirty propagation, so as a
+		// precaution we destroy the history on the UI thread, where this would
+		// be OK.
+		ParallelAlgo::callOnUIThread(
+			[history] () {}
+		);
+	}
+
+);
+
+}  // namespace
+
+SetMembershipInspector::SetMembershipInspector(
+    const GafferScene::ScenePlugPtr &scene,
+    const Gaffer::PlugPtr &editScope,
+    IECore::InternedString setName,
+    const std::string &name
+) :
+Inspector( "setMembership", name == "" ? setName.string() : name, editScope ),
+m_scene( scene ),
+m_setName( setName )
+{
+
+}
+
+GafferScene::SceneAlgo::History::ConstPtr SetMembershipInspector::history() const
+{
+    if( !m_scene->existsPlug()->getValue() )
+    {
+        return nullptr;
+    }
+
+    return g_historyCache.get( HistoryCacheKey( m_scene->objectPlug() ), Context::current()->canceller() );
+}
+
+IECore::ConstObjectPtr SetMembershipInspector::value( const GafferScene::SceneAlgo::History *history ) const
+{
+    if( history )
+    {
+        const auto &path = history->context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
+        const PathMatcher setMembers = SetAlgo::evaluateSetExpression( m_setName.string(), history->scene.get() );
+        
+        if( setMembers.match( path ) & ( PathMatcher::Result::AncestorMatch | PathMatcher::Result::ExactMatch ) )
+        {
+            return new IntData( setMembers.match( path ) );
+        }
+    }
+    return nullptr;
+}
+
+Gaffer::ValuePlugPtr SetMembershipInspector::source( const GafferScene::SceneAlgo::History *history, std::string &editWarning ) const
+{
+    auto sceneNode = runTimeCast<SceneNode>( history->scene->node() );
+    if( !sceneNode || history->scene != sceneNode->outPlug() || !sceneNode->enabledPlug()->getValue() )
+    {
+        return nullptr;
+    }
+
+    if( const auto objectSource = runTimeCast<ObjectSource>( sceneNode ) )
+    {
+        if( StringAlgo::matchMultiple( m_setName, objectSource->setsPlug()->getValue() ) )
+        {
+            return objectSource->setsPlug();
+        }
+    }
+    if( const auto setSource = runTimeCast<GafferScene::Set>( sceneNode ) )
+    {
+        if( auto nameSource = setSource->namePlug()->source<StringVectorDataPlug>() )
+        {
+            const std::vector<std::string> &names = nameSource->getValue()->readable();
+            if( std::find( names.begin(), names.end(), m_setName.string() ) == names.end() )
+            {
+                return nullptr;
+            }
+            if( const auto spreadsheetSource = runTimeCast<Spreadsheet>( nameSource->parent() ) )
+            {
+                return spreadsheetSource->rowsPlug()->row( m_setName );
+            }
```

Comment:
This is what makes us compatible with EditScopeAlgo, right? Should we also be validating that the rows plug is actually driving the filter for the Set node? And maybe do that in such a way that simple user-made "Set + PathFilter" setups are also supported?